src/graphql/types/object/btc-wallet.ts

```diff
@@ -43,8 +44,26 @@ const BtcWallet = GT.Object<Wallet>({
     },
     transactions: {
       type: TransactionConnection,
-      args: connectionArgs,
+      args: { ...connectionArgs, addresses: { type: GT.List(OnChainAddress) } },
       resolve: async (source, args) => {
+        let { addresses } = args
+        if (addresses) {
+          addresses = addresses.length ? addresses : source.onChainAddresses()
+          for (const address of addresses) {
+            if (address instanceof Error) throw address
+          }
+
+          const { result: transactions, error } =
+            await Wallets.getTransactionsForWalletsByAddresses({
+              wallets: [source],
+              addresses,
+            })
+          if (error instanceof Error || transactions === null) {
+            throw error
```

Comment:
does this needs to use mapError?

Reply:
Fixed: https://github.com/GaloyMoney/galoy/pull/1511/commits/5e625b2536da9d0e80b0064b3b505e80fb21c125